src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/tests/test_custom.py

```diff
@@ -138,3 +138,56 @@ def test_merge_credentials(self):
 
         os.remove(existing.name)
         os.remove(addition.name)
+
+    def test_merge_credentials_already_present(self):
+        existing = tempfile.NamedTemporaryFile(delete=False)
+        existing.close()
+        addition = tempfile.NamedTemporaryFile(delete=False)
+        addition.close()
+        obj1 = {
+            'clusters': [
+                'cluster1',
+                'cluster2'
+            ],
+            'contexts': [
+                'context1',
+                'context2'
+            ],
+            'users': [
+                'user1',
+                'user2'
+            ],
+            'current-context': 'cluster1',
+        }
+        with open(existing.name, 'w+') as stream:
+            yaml.dump(obj1, stream)
+
+        obj2 = {
+            'clusters': [
+                'cluster2'
+            ],
+            'contexts': [
+                'context2'
+            ],
+            'users': [
+                'user2'
+            ],
+            'current-context': 'cluster2',
+        }
+
+        with open(addition.name, 'w+') as stream:
+            yaml.dump(obj2, stream)
+        merge_kubernetes_configurations(existing.name, addition.name)
+
+        with open(existing.name, 'r') as stream:
+            merged = yaml.load(stream)
+        self.assertEqual(len(merged['clusters']), 2)
+        self.assertEqual(merged['clusters'], ['cluster1', 'cluster2'])
+        self.assertEqual(len(merged['contexts']), 2)
+        self.assertEqual(merged['contexts'], ['context1', 'context2'])
+        self.assertEqual(len(merged['users']), 2)
+        self.assertEqual(merged['users'], ['user1', 'user2'])
+        self.assertEqual(merged['current-context'], obj2['current-context'])
+
+        os.remove(existing.name)
```

Comment:
When the test fails, will this be executed? Otherwise, will the file be left behind? It doesn't cause huge trouble